### PR TITLE
Configure ports using test.config

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses "that mess with ports".

Proposed changes include:
* Remove all hard-coded magic ports from test suites code.
* Uniform port names (same names for ports in test.config and var files).
* Use separate variables for ports in templates (do not put ports into other variables).
* Uniform naming (i.e. `_port` suffix).
* Remove `ejabberd_` prefix from some port variables.
* Move node_name on top of vars configs.

General idea:
* If you need to introduce new port - add it to var file of the corresponding test node and into test.config.
* Use the same name and value for both of them.

Reasoning:
* It would make possible to process port variables to assign ports dynamically.
* We can remove ports from test.config later, if we generate `hosts` automatically from var files.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[7245.1](https://travis-ci.org/esl/MongooseIM/jobs/588907326) / Erlang 22.0 / small_tests / 5c9b5bdb6194dddacbd07ad4c127ea5c1f22e70a
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2475/7245/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2475/7245/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-09-24_11.41.33/lib.mongooseim.logs/run.2019-09-24_11.41.33/suite.log.html)

File: big_tests/run_common_test.erl [hunk 1]
```diff
@@ -270,7 +270,8 @@ enable_preset_on_node(Node, PresetVars, HostVars) ->
     NewVars = lists:foldl(fun ({Var, Val}, Acc) ->
                               lists:keystore(Var, 1, Acc, {Var, Val})
                           end, Default, PresetVars),
-    NewCfgFile = bbmustache:render(Template, NewVars, [{key_type, atom}]),
+    Tmp = bbmustache:render(Template, NewVars, [{key_type, atom}]),
+    NewCfgFile = bbmustache:render(Tmp, NewVars, [{key_type, atom}]),
```
Comment (NelsonVides) [inline big_tests/run_common_test.erl]: It's quite confusing to render twice, so maybe leave here a comment as well, one of those `%% Render twice to replace variables in variables`

File: big_tests/tests/push_http_SUITE.erl [hunk 2]
```diff
@@ -165,6 +165,12 @@ stop_pool() ->
     ejabberd_node_utils:call_fun(mongoose_wpool, stop, [http, <<"localhost">>, http_pool]),
     ok.
 
+http_notifications_port() ->
+    ct:get_config({hosts, mim, http_notifications_port}).
+
+http_notifications_host() ->
+    "http://localhost:" ++ integer_to_list(http_notifications_port()).
+
```
Comment (NelsonVides) [inline big_tests/tests/push_http_SUITE.erl]: This is duplicated between `mod_http_notification_SUITE` and `push_http_SUITE`, wouldn't it make sense to move it to some helper module? Not like crucial, but it wouldn't hurt, right?

File: big_tests/tests/sasl_external_SUITE.erl [hunk 3]
```diff
@@ -317,6 +317,7 @@ self_signed_cert_is_allowed_with(EscalusTransport, C) ->
 no_cert_fails_to_authenticate(_C) ->
     UserSpec = [{username, <<"no_cert_user">>},
 		{server, <<"localhost">>},
+        {port, ct:get_config({hosts, mim, c2s_port})},
```
Comment (NelsonVides) [inline big_tests/tests/sasl_external_SUITE.erl]: Broken indentation?

Comment (arcusfelis) [reply big_tests/tests/sasl_external_SUITE.erl]: kinda.
There is a mix of tabs and whitespaces across this file.
I haven't reindent the file to reduce size of this PR.

(tldr: one tab is 4 spaces here)

File: big_tests/tests/push_http_SUITE.erl (hunk 2 above)
Comment (arcusfelis) [reply big_tests/tests/push_http_SUITE.erl]: Done

[configure-ports-using-test-config 3c61bf0e0] Move http_notifications_port/0, http_notifications_host/0 into push_helper
 3 files changed, 12 insertions(+), 12 deletions(-)

File: big_tests/tests/sasl_external_SUITE.erl (hunk 3 above)
Comment (arcusfelis) [reply big_tests/tests/sasl_external_SUITE.erl]: The whole file needs re-indention. 
I would not do it in this PR to simplify code review.
We can reindent in a separate PR with skip-ci.

**Comment by mongoose-im** [comment]
[7267.1](https://travis-ci.org/esl/MongooseIM/jobs/589035366) / Erlang 22.0 / small_tests / face47e313e35fc98d15e82bd9151388bd57c21e
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2475/7267/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2475/7267/small_tests.22.0/small/ct_run.mongooseim@localhost.2019-09-24_18.19.55/lib.mongooseim.logs/run.2019-09-24_18.19.56/suite.log.html)

**Pull request merged**